data-prepper-api/src/main/java/org/opensearch/dataprepper/model/schemas/IfPresentAlsoRequire.java

```diff
@@ -0,0 +1,21 @@
+package org.opensearch.dataprepper.model.schemas;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marker annotation used in schema generation to define the names and corresponding values of other required
+ * properties to be returned, if the property represented by the annotated field/method is present.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface IfPresentAlsoRequire {
```

Comment:
Schemas and validations are tightly related. So we should aim to have a common language for both schemas and validations. Otherwise the developers will need to track schemas and validations independently and this may introduce misses.

I think a few modifications to the documentation and package will help here.

What do you think about using the name `@AlsoRequires()`?

Reply:
Unfortunately I am not able to find an out-of-the-box annotation from Jakarta or JsonProperty to achieve this schema property. Thus the custom annotation. I agree the tradeoff would be to maintain a set of custom annotation for schema only. 
>So we should aim to have a common language for both schemas and validations.
Could you specify what you mean here?

Reply:
I believe you are correct - Jakarta does not have this annotation.

However, you are adding an annotation that you expect to be useful for schema generation. I suggest that we take a different approach. You should be adding annotations that let plugin developers express constraints in one place. Data Prepper should be able to use these annotations for both schema generation and for validation.

So:
1. We need a new annotation because there is not one available.
2. We should design it to support validation.
3. We should document it as being able to express configuration constraints, not as simply a schema generation tool.

Reply:
Thanks! I will open a separate issue for that. To incorporate validation requires additional work in pipeline parsing.

Reply:
https://github.com/opensearch-project/data-prepper/issues/4935